client/blocks/credit-card/payment-method.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { Elements, useStripe } from '@stripe/react-stripe-js';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getStripeServerData } from '../stripe-utils';
+import { useCheckoutSubscriptions } from './use-checkout-subscriptions';
+import { InlineCard, CardElements } from './elements';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+export const getStripeCreditCardIcons = () => {
+	return Object.entries( getStripeServerData().icons ).map(
+		( [ id, { src, alt } ] ) => {
+			return {
+				id,
+				src,
+				alt,
+			};
+		}
+	);
+};
+
+/**
+ * Stripe Credit Card component
+ *
+ * @param {RegisteredPaymentMethodProps} props Incoming props
+ */
+const CreditCardComponent = ( {
+	billing,
+	eventRegistration,
+	emitResponse,
+	components,
+} ) => {
+	const { ValidationInputError, PaymentMethodIcons } = components;
+	const [ sourceId, setSourceId ] = useState( '' );
+	const stripe = useStripe();
+	const onStripeError = useCheckoutSubscriptions(
+		eventRegistration,
+		billing,
+		sourceId,
+		setSourceId,
+		emitResponse,
+		stripe
+	);
+	const onChange = ( paymentEvent ) => {
+		if ( paymentEvent.error ) {
+			onStripeError( paymentEvent );
+		}
+		setSourceId( '' );
+	};
+	const cardIcons = getStripeCreditCardIcons();
+
+	const renderedCardElement = getStripeServerData().inline_cc_form ? (
+		<InlineCard
+			onChange={ onChange }
+			inputErrorComponent={ ValidationInputError }
+		/>
+	) : (
+		<CardElements
+			onChange={ onChange }
+			inputErrorComponent={ ValidationInputError }
+		/>
+	);
+	return (
+		<>
+			{ renderedCardElement }
+			{ PaymentMethodIcons && cardIcons.length && (
+				<PaymentMethodIcons icons={ cardIcons } align="left" />
+			) }
+		</>
+	);
+};
+
+export const StripeCreditCard = ( props ) => {
+	// Make sure `locale` defaults to 'en_US' if it's not found in the server provided
+	// configuration.
+	const { locale = 'en_US' } = getStripeServerData().button;
+	const { stripe } = props;
+
+	return (
+		<Elements stripe={ stripe } locale={ locale }>
```

Comment:
> Also, `en_US` doesn't seem to be a supported locale.

😱 
I'm pretty sure we send `en_US` from the server to the frontend by default.

I'll change this to `en` just in case though. We definitely need to look into this some more.

Reply:
I removed the locale on the Stripe `<Elements >` in 8f4a93c3ace45fde83119f24212ab2af9580049e

Reply:
@reykjalin - wanted to double-check this. Have you been able to test translations in different locales?

Reply:
We don't have any translations available for our part of the plugin, but the rest of the blocks are translated. Looks a bit wonky:

![image](https://user-images.githubusercontent.com/13835680/118203294-44844c80-b44b-11eb-93d8-2341090ac5d6.png)

It's a bit weird to see some of the Stripe parts localized but other parts are not 😬 

I don't know how add test translations to see if the JS translations are loaded. I suspect we may need to do something like @diegocurbelo did for WCPay in https://github.com/Automattic/woocommerce-payments/pull/1716

It's difficult to know whether this is the case or not though. Can I somehow create a custom translations file and upload it to my test install? @diegocurbelo did you have to do something like that during your work on https://github.com/Automattic/woocommerce-payments/pull/1716?

Reply:
@reykjalin I think the parts not translated are generated by Stripe. If you add a locale parameter to the Stripe object, it should work.

```JavaScript
var stripe = Stripe('pk_test_••••••••••', { locale: 'es' });
```

But it needs to be a [supported locale](https://stripe.com/docs/js/appendix/supported_locales).

Reply:
@reykjalin I think the parts not translated are generated by Stripe. If you add a locale parameter to the Stripe object, it should work.

```JavaScript
const stripe = Stripe( 'pk_test_••••••••••',
    {
        locale: 'es'
    }
);
```

But it needs to be a [supported locale](https://stripe.com/docs/js/appendix/supported_locales).

Reply:
> If you add a locale parameter to the Stripe object

Do we send the store's locale in this object? We do it for the shortcode checkout. From Ricardo's comment on Slack, sounds like this will need to be added (possibly in a separate PR).

Reply:
I thought we added the locale already, but it seems that's not the case. In any case, that's more for the error messages, the `Credit Card Information` placeholder is set by us [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/3c8f3c5cbc0ca862e5499535ffa2f1d349b59ac8/client/blocks/credit-card/elements.js#L57-L62).

I'll fix the locale, but most of the wonky UX stuff here is related to our translation strings, not the Stripe locale.

Reply:
I created a custom translations file, tried uploading the translations, and checked if they appeared, and it looks like everything is in order 🎉 

![image](https://user-images.githubusercontent.com/13835680/118290459-fd876d00-b4c5-11eb-8e24-e5617cb5c3b6.png)

To reproduce this test:

```sh
cd languages
cp woocommerce-gateway-stripe.pot woocommerce-gateway-stripe-es_ES.po

# Now open 'woocommerce-gateway-stripe-es_ES.po' and add a translation for
# any string you know how to find in the Blocks (I chose 'Credit Card Information')
# and add a translation (in my case that's 'Información de la tarjeta de crédito').

vi woocommerce-gateway-stripe-es_ES.po

# Once that's done, you can generate the json files:

wp i18n make-json woocommerce-gateway-stripe-es_ES.po --no-purge

# Now you should have a bunch of 'woocommerce-gateway-stripe-es_ES-<hash>.json
# files in your directory.

$ ls
woocommerce-gateway-stripe-es_ES-580df76e4e40f57ad925a679d96cc41b.json
woocommerce-gateway-stripe-es_ES-bb0426d856ee6e77f3d431a9fdf1ffae.json
woocommerce-gateway-stripe-es_ES-d25e096d7473e1ff41992c061f48b0af.json
woocommerce-gateway-stripe-es_ES-dfbff627e6c248bcb3b61d7d06da9ca9.json
woocommerce-gateway-stripe-es_ES-wc-stripe-blocks-integration.json
woocommerce-gateway-stripe-es_ES.po
woocommerce-gateway-stripe.pot

# Now copy those json files to the 'wp-content/languages/plugins' directory
# on the server.
# In my setup my code was already located in a WordPress installation,
# so I was able to copy these directly; you will need to do this differently
# with our Docker setup.
cp plugins/woocommerce-gateway-stripe/languages/*.json languages/plugins/
```

Reply:
I've added the store's locale to Stripe's initialization with e202847eca19ba6283d9fa099a05975f9fcecf67. I made sure to normalize the locale for Stripe's list, so there shouldn't be any issues.

There's still the question of whether we should be doing this. Stripe defaults to the browser's locale, so maybe we should be relying on that instead?
The shortcode flow uses the browser's locale, if we're changing this in the blocks checkout we should do the same for the shortcode flow — that's now easy through the `wc_stripe_params.stripe_locale` value, which is available both for blocks and the shortcode flows.